js/modules/inputValidation.js

```diff
@@ -0,0 +1,108 @@
+const forms = Array.from(document.forms);
+let isInputHandlerAdded = false;
+
+const REGEXP = {
+	NAME: /^[а-яА-ЯёЁa-zA-ZЁёЇїІіЄєҐґ']+$/,
+	PHONE: /^(?! )^[+\-\s()0-9]{10,20}$/,
+}
+
+const RULES = [
+	{
+		name: "name",
+		validate: function(input){
+
+			return REGEXP.NAME.test(input.value);
+		}
+	},
+	{
+		name: "phone",
+		validate: function(input){
+
+			return REGEXP.PHONE.test(input.value);
+		}
+	},
+	{
+		name: "checkbox",
+		validate: function(input){
+
+			return input.checked;
+		}
+	}
+]
+
+document.onsubmit = () => {return false};
+
+function validateField(input){
+	let checkings = [];
+
+	RULES.forEach((rule) => {
+		if(Object.hasOwn(input.dataset, rule.name)){
+			const isValid = rule.validate(input);
+			checkings.push(isValid);
+		}
+	});
+	return checkings.every((item) => item);
+}
+
+function isFormValid(form){
+	const checkings = [];
+	const filteredFields = Array.from(form.elements).filter((element)=>{
+		return Object.hasOwn(element.dataset, "toValidate");
+	});
+
+	filteredFields.forEach((element)=> {
+
+		const isValid = validateField(element);
+
+		if(!isInputHandlerAdded) {
+			addInputHandler(element);
+		}
+		
+		checkings.push(isValid);
+
+		if(!isValid) {
+			showError(element);
+		} else {
+			hideError(element);
+		}
+	})
+	isInputHandlerAdded = true;
+	
+	return checkings.every((item) => item);
+}
+
+forms.forEach((form) => form.addEventListener("submit", handleFormSubmit));
```

Comment:
put this string above the functions' initializations

Reply:
Done.